pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -315,6 +324,25 @@ loop:
 		}
 	}
 
+	// if we have restricted-v2, and we're not allowed (this means restricted-v2 did not match) and the user cannot use restricted-v1
+	// then we should check to see if restricted-v1 would allow the pod.  If so, prepend a specific failure message.
+	userCannotUseForRestricted := sets.NewString(denied...).Has("restricted")
+	hasRestrictedV2 := restrictedV2SCCProvider != nil
+	isAllowed := allowingProvider != nil
+	if hasRestrictedV2 && !isAllowed && userCannotUseForRestricted {
+		_, restrictedErrs := appliesToPod(restrictedSCCProvider, pod)
```

Comment:
Could it happen that restrictedSCCProvider is not set?

Reply:
No.  In order to be denied "restricted", the restricted SCC was iterated over in this loop `for i, provider = range providers {`

Reply:
comment added.